google/cloud/bigquery_storage_v1/writer.py

```diff
@@ -263,50 +153,17 @@ def send(self, request: gapic_types.AppendRowsRequest) -> "AppendRowsFuture":
             A future, which can be used to process the response when it
             arrives.
         """
-        if self._closed:
-            raise bqstorage_exceptions.StreamClosedError(
-                "This manager has been closed and can not be used."
-            )
-
-        # If the manager hasn't been openned yet, automatically open it. Only
-        # one call to `send()` should attempt to open the RPC. After `_open()`,
-        # the stream is active, unless something went wrong with the first call
-        # to open, in which case this send will fail anyway due to a closed
-        # RPC.
-        with self._opening:
-            if not self.is_active:
-                return self._open(request)
-
-        # For each request, we expect exactly one response (in order). Add a
-        # future to the queue so that when the response comes, the callback can
-        # pull it off and notify completion.
-        future = AppendRowsFuture(self)
-        self._futures_queue.put(future)
-        self._rpc.send(request)
-        return future
-
-    def _on_response(self, response: gapic_types.AppendRowsResponse):
-        """Process a response from a consumer callback."""
-        # If the stream has closed, but somehow we still got a response message
-        # back, discard it. The response futures queue has been drained, with
-        # an exception reported.
-        if self._closed:
-            raise bqstorage_exceptions.StreamClosedError(
-                f"Stream closed before receiving response: {response}"
+        if self._stream_name is None:
+            self._stream_name = request.write_stream
+        elif request.write_stream != self._stream_name:
+            raise ValueError(
+                "Stream name is already set by the original request as "
+                f"{self._stream_name}, different from {request.write_stream} "
+                "in this request. Please use the same name or open a new stream."
             )
+        return self._connection.send(request)
```

Comment:
read lock needed?

Reply:
I think in our case it would be fine, because the concurrency library that we use (threading) supports only one CPU.